### PR TITLE
Close #1772 Add GuestCardClass to Variant

### LINKED ISSUE #1772: Add GuestCardClass to Variant
- [x] Next, connect the color field in GuestCard to Variant so that it can fetch all data from GuestCard.
Screenshot from 2024-08-08 14-55-40

image

### PR DESCRIPTION
**Pull request by Sreyleak-Deth**
Screenshot from 2024-08-15 11-35-41

image

### CODE REVIEW

**Comment by channainfo** [comment]
@Sreyleak-Deth let's follow the UI in the Google Wallet. Let's not add the field to the variant.

File: db/migrate/20240815012429_add_guest_card_class_to_variant.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddGuestCardClassToVariant < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :spree_variants, :guest_card_class, foreign_key: { to_table: :cm_guest_card_classes }, index: true, if_not_exists: true
```
Comment (channainfo) [inline db/migrate/20240815012429_add_guest_card_class_to_variant.rb:3]: It is not always the case to have guest_card_class in the variant. Let's not add it here. If the guest_card_class is reusable let's associate it with another table.

File: app/overrides/spree/admin/variants/_form/guest_card_class.html.erb.deface [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<!-- insert_bottom "[data-hook='admin_variant_form_additional_fields']" -->
+
+<div class="form-group" data-hook="guest_card_classes">
+  <%= f.field_container :guest_card_class_id do %>
+    <%= f.label :guest_card_class_id, Spree.t('guest_card_classes') %>
+    <%= f.collection_select :guest_card_class_id, SpreeCmCommissioner::GuestCardClass.all, :id, :name, { prompt: Spree.t('match_choices.none') }, { class: 'form-control' } %>
```
Comment (channainfo) [inline app/overrides/spree/admin/variants/_form/guest_card_class.html.erb.deface:6]: Let's integrate it similarly to how Google Wallet is set up.

**Review by channainfo — CHANGES_REQUESTED** [review]

File: app/overrides/spree/admin/variants/_form/guest_card_class.html.erb.deface (hunk 2 above)
Comment (Sreyleak-Deth) [reply app/overrides/spree/admin/variants/_form/guest_card_class.html.erb.deface:6]: > Let's integrate it similarly to how Google Wallet is set up.

Noted bong, i will resolve it bong, Thanks

**Pull request closed without merging**